include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.hpp

```diff
@@ -239,7 +245,20 @@ class COE
     /// @param aCartesianState A cartesian state
     /// @param aGravitationalParameter A gravitational parameter
     /// @return COE
-    static COE Cartesian(const COE::CartesianState& aCartesianState, const Derived& aGravitationalParameter);
+    static COE Cartesian(
+        const COE::CartesianState& aCartesianState,
+        const Derived& aGravitationalParameter = EarthGravitationalModel::EGM2008.gravitationalParameter_
+    );
+
+    /// @brief Construct a COE from a State
+    ///
+    /// @param aState A state
+    /// @param aGravitationalParameter A gravitational parameter
+    /// @return COE
+    static COE FromState(
+        const State& aState,
+        const Derived& aGravitationalParameter = EarthGravitationalModel::EGM2008.gravitationalParameter_
+    );
```

Comment:
This feels like a little bit of a specific ad-hoc solution here.
I think a more generalizable solution could be to delete the `typedef Pair<Position, Velocity> CartesianState;` and replace the `COE::Cartesian` factory method with this new `COE::FromState` factory method you added. That way coordinate subset and reference frame checking are actually undertaken instead of assumptions made when parsing the pair of position and velocity like happens [here](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/434/files#diff-e928c48a82575ab8a2b99e3fa061796625187c58c18542ac932cfde16a4ca02dR449-R450)

Reply:
I think that constructors should accept the minimum required information to create the object, and then we can have named constructors that simplify interfaces. There's no reason we can't have both. Also it would be a pretty big breaking change :)